File: src/main/java/me/alegian/thavma/impl/common/menu/slot/DynamicSlot.java
package me.alegian.thavma.impl.common.menu.slot;

import me.alegian.thavma.impl.common.menu.Menu;

/**
 * Used for adding size to slots, allowing them to be dynamically positioned
 */
public interface DynamicSlot<T extends Menu> {
  int getX();

  int getY();

  void setX(int x);

  void setY(int y);

  int getSize();

  T getMenu();
}

File: src/main/java/me/alegian/thavma/impl/common/menu/slot/DynamicSlot.kt
package me.alegian.thavma.impl.common.menu.slot

import me.alegian.thavma.impl.common.menu.Menu

/**
 * Used for adding size to slots, allowing them to be dynamically positioned
 */
interface DynamicSlot<T : Menu> {
  var actualX: Int

  var actualY: Int

  var size: Int

  val menu: T
}
